File: widget_bridge/trunk/src/js/main/js/topcoder/widgets/bridge/UploadedDocument.js
/*
 * Copyright (C) 2008 TopCoder Inc., All Rights Reserved.
 */


/**
 * <p>This Java Script object represents the Uploaded Document data.
 *
 * Thread safety:  Mutable and not thread-safe.</p>
 *
 * Note: 
 *   the fileName means the local file name that user wants to upload. It will be
  *  used to generate a input-file html control when uploading the file.  And it
  *  won't be generated into the json string.
 * 
 * @author Standlove, pinoydream
 * @version 1.0
 *
 */

/**
 * <p>Construct the js object with the json object.
 * 
 * Set json's properties to the corresponding properties of this js object.</p>
 *
 * <p>If this constructor is called without a parameter, it serves as an empty default constructor.</p>
 */ 
js.topcoder.widgets.bridge.UploadedDocument = function (/* JSON Object */ json) {
	/**
	 * This is to make the object available to private methods.
	 */
	var that = this;

	/**
	 * <p>Represents the documentId</p>
	 * <p>Initial Value: -1, means that is not set</p>
	 * <p>Accessed In: getter method</p>
	 * <p>Modified In: setter method</p>
	 * <p>Utilized In: none</p>
	 * <p>Valid Values: can be -1, can be greater or equal to 0</p>
	 */
    var documentID /* long */ = -1;

	/**
	 * <p>Represents the contestId</p>
	 * <p>Initial Value: -1, means that is not set</p>
	 * <p>Accessed In: getter method</p>
	 * <p>Modified In: setter method</p>
	 * <p>Utilized In: none</p>
	 * <p>Valid Values: can be -1, can be greater or equal to 0</p>
	 */
    var contestID /* long */ = -1;

	/**
	 * <p>Represents the fileName</p>
	 * <p>Initial Value: null, means that is not set</p>
	 * <p>Accessed In: getter method</p>
	 * <p>Modified In: setter method</p>
	 * <p>Utilized In: none</p>
	 * <p>Valid Values: all values, can be null, can be empty</p>
	 */
    var fileName /* String */ = null;

	/**
	 * <p>Represents the path</p>
	 * <p>Initial Value: null, means that is not set</p>
	 * <p>Accessed In: getter method</p>
	 * <p>Modified In: setter method</p>
	 * <p>Utilized In: none</p>
	 * <p>Valid Values: all values, can be null, can be empty</p>
	 */
    var path /* String */ = null;

	/**
	 * <p>Represents the description</p>
	 * <p>Initial Value: null, means that is not set</p>
	 * <p>Accessed In: getter method</p>
	 * <p>Modified In: setter method</p>
	 * <p>Utilized In: none</p>
	 * <p>Valid Values: all values, can be null, can be empty</p>
	 */
    var description /* String */  = null;

	/**
	 * <p>Construct the js object with the json object.
	 * Set json's properties to the corresponding properties of this js object.
	 *
	 * Note: we can't throw IllegalArgumentException here since we will
	 * support an empty Constructor for this class.
	 *
	 * Note: Setter Methods can't actually be used here since it doesn't
	 * retain the values of the parameters after this object has been created.
	 * The Getter methods won't give the expected result and will return 'undefined' or 'null'.
	 * This is an issue with the use of 'this'. 
	 */ 
	if (json != null) {
		// set the document id
		if (typeof(json.documentID) != "undefined" && typeof(json.documentID) == "number") {		
			that.documentID = json.documentID;
		} else {
			throw new js.topcoder.widgets.bridge.IllegalArgumentException("json.documentID","json.documentID does not exist");
		}		
		// set the contest id
		if (typeof(json.contestID) != "undefined" && typeof(json.contestID) == "number") {		
			that.contestID = json.contestID;
		} else {
			throw new js.topcoder.widgets.bridge.IllegalArgumentException("json.contestID","json.contestID does not exist");
		}			
		// set the description
		if (typeof(json.description) != "undefined" && typeof(json.description) == "string") {		
			that.description = json.description;
		} else {
			throw new js.topcoder.widgets.bridge.IllegalArgumentException("json.description","json.description does not exist");
		}			
		// set the file name
		if (typeof(json.fileName) != "undefined" && typeof(json.fileName) == "string") {		
			that.fileName = json.fileName;
		} else {
			throw new js.topcoder.widgets.bridge.IllegalArgumentException("json.fileName","json.fileName does not exist");
		}			
		// set the path
		if (typeof(json.path) != "undefined" && typeof(json.path) == "string") {		
			that.path = json.path;
		} else {
			throw new js.topcoder.widgets.bridge.IllegalArgumentException("json.path","json.path does not exist");
		}			
	}

    /**
     * <p>Returns the contestID.</p>
     */
    this.getContestID = function /* long */ () {
        return that.contestID;
    }

    /**
     * <p>Sets the contestID.</p>
     */
    this.setContestID = function /* void */ (/* long */ contestID) {
        that.contestID = contestID;
    }

    /**
     * <p>Returns the description.</p>
     */
    this.getDescription = function /* String */ () {
        return that.description;
    }

    /**
     * <p>Sets the description.</p>
     */
    this.setDescription = function /* void */ (/* String */ description) {
        that.description = description;
    }

    /**
     * <p>Returns the documentID.</p>
     */
    this.getDocumentID = function /* long */ () {
        return that.documentID;
    }

    /**
     * <p>Sets the documentID.</p>
     */
    this.setDocumentID = function /* void */ (/* long */ documentID) {
        that.documentID = documentID;
    }

    /**
     * <p>Returns the fileName.</p>
     */
    this.getFileName = function /* String */ () {
        return that.fileName;
    }

    /**
     * <p>Sets the fileName.</p>
     */
    this.setFileName = function /* void */ (/* String */ fileName) {
        that.fileName = fileName;
    }

    /**
     * <p>Returns the path.</p>
     */
    this.getPath = function /* String */ () {
        return that.path;
    }

    /**
     * <p>Sets the path.</p>
     */
    this.setPath = function /* void */ (/* String */ path) {
        that.path = path;
    }

	/**
	 * <p>Convert this JavaScript object to a JSON string and return.
	 *
	 * It will generate json string in the following format: (The $xxx should be replaced by corresponding variables.)</p>
	 *
	 * NOTE: the string-value needs to be enclosed by double-quotes according to JSON format.  Properties with null
	 * value should also be added to the generated json string with null value. 
	 *
	 * {"fileName" : $fileName,"documentID" : $documentID,"contestID" : $contestID,"description" : "$description"}
	 */
	this.toJSON = function /* String */ () {
		return "{\"fileName\" : \""+that.getFileName()+"\",\"path\" :\""+that.getPath()+"\",\"documentID\" : "+that.getDocumentID()+
			",\"contestID\" : "+that.getContestID()+", \"description\" : \""+that.getDescription()+"\"}";
	}

} // end

File: studio_service/branches/multi_subviewer/src/java/main/com/topcoder/service/studio/UploadedDocument.java
/*
 * Copyright (C) 2009 TopCoder Inc., All Rights Reserved.
 */
package com.topcoder.service.studio;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlType;
import java.io.Serializable;

/**
 * <p>
 * It is the DTO class which is used to transfer document and document content
 * data. The information can be null or can be empty, therefore this check is
 * not present in the setters. It's the related to the equivalent Document
 * entity.
 * </p>
 *
 * <p>
 * This class is not thread safe because it's highly mutable
 * </p>
 *
 * @author fabrizyo, TCSDEVELOPER
 * @version 1.0
 */
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "uploadedDocument", propOrder = { "documentId", "contestId", "file", "description",
        "fileName", "documentTypeId", "mimeTypeId", "path" })
public class UploadedDocument implements Serializable {
    /**
     * <p>
     * Represents the document Id.
     * </p>
     */
    private long documentId = -1;

    /**
     * <p>
     * Represents the contest Id.
     * </p>
     */
    private long contestId = -1;

    /**
     * <p>
     * Represents the file content.
     * </p>
     */
    private byte[] file;

    /**
     * <p>
     * Represents the description of document type.
     * </p>
     */
    private String description;

    /**
     * <p>
     * Represents the fileName.
     * </p>
     */
    private String fileName;

    /**
     * <p>
     * Represents the document Type Id.
     * </p>
     */
    private long documentTypeId = -1;

    /**
     * <p>
     * Represents the mimeType Id.
     * </p>
     */
    private long mimeTypeId = -1;

    /**
     * <p>
     * Represents the document path.
     * </p>
     */
    private String path;

    /**
     * <p>
     * This is the default constructor. It does nothing.
     * </p>
     */
    public UploadedDocument() {
    }

    /**
     * <p>
     * Return the documentId.
     * </p>
     *
     * @return the documentId
     */
    public long getDocumentId() {
        return documentId;
    }

    /**
     * <p>
     * Set the documentId.
     * </p>
     *
     * @param documentId the documentId to set
     */
    public void setDocumentId(long documentId) {
        this.documentId = documentId;
    }

    /**
     * <p>
     * Return the contestId.
     * </p>
     *
     * @return the contestId
     */
    public long getContestId() {
        return contestId;
    }

    /**
     * <p>
     * Set the contestId.
     * </p>
     *
     * @param contestId the contestId to set
     */
    public void setContestId(long contestId) {
        this.contestId = contestId;
    }

    /**
     * <p>
     * Return the file.Return the reference, to save a bit of memory, a file
     * could be large.
     * </p>
     *
     * @return the file.
     */
    public byte[] getFile() {
        return file;
    }

    /**
     * <p>
     * Set the file. Set the reference, to save a bit of memory, a file could be
     * large.
     * </p>
     *
     * @param file the file to set
     */
    public void setFile(byte[] file) {
        this.file = file;
    }

    /**
     * <p>
     * Return the description.
     * </p>
     *
     * @return the description.
     */
    public String getDescription() {
        return description;
    }

    /**
     * <p>
     * Set the description.
     * </p>
     *
     * @param description the description to set.
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * <p>
     * Return the fileName.
     * </p>
     *
     * @return the fileName.
     */
    public String getFileName() {
        return fileName;
    }

    /**
     * <p>
     * Set the fileName.
     * </p>
     *
     * @param fileName the fileName to set.
     */
    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    /**
     * <p>
     * Return the documentTypeId.
     * </p>
     *
     * @return the documentTypeId.
     */
    public long getDocumentTypeId() {
        return documentTypeId;
    }

    /**
     * <p>
     * Set the documentTypeId.
     * </p>
     *
     * @param documentTypeId the documentTypeId to set.
     */
    public void setDocumentTypeId(long documentTypeId) {
        this.documentTypeId = documentTypeId;
    }

    /**
     * <p>
     * Return the mimeTypeId.
     * </p>
     *
     * @return the mimeTypeId.
     */
    public long getMimeTypeId() {
        return mimeTypeId;
    }

    /**
     * <p>
     * Set the mimeTypeId.
     * </p>
     *
     * @param mimeTypeId the mimeTypeId to set.
     */
    public void setMimeTypeId(long mimeTypeId) {
        this.mimeTypeId = mimeTypeId;
    }

    /**
     * <p>
     * Return the path.
     * </p>
     *
     * @return the path.
     */
    public String getPath() {
        return path;
    }

    /**
     * <p>
     * Set the path.
     * </p>
     *
     * @param path the path to set.
     */
    public void setPath(String path) {
        this.path = path;
    }
}
